tasks/kernel_matrix_testing/infra.py

```diff
@@ -1,26 +1,35 @@
+from __future__ import annotations
+
 import glob
 import json
 import os
+from typing import TYPE_CHECKING, Dict, List, Optional
+
+from invoke.context import Context
 
 from tasks.kernel_matrix_testing.kmt_os import get_kmt_os
 from tasks.kernel_matrix_testing.tool import Exit, ask, error
 
+if TYPE_CHECKING:
+    from tasks.kernel_matrix_testing.types import ArchOrLocal, PathOrStr
+
 
 class LocalCommandRunner:
     @staticmethod
-    def run_cmd(ctx, _, cmd, allow_fail, verbose):
+    def run_cmd(ctx: Context, _: 'HostInstance', cmd: str, allow_fail: bool, verbose: bool):
         res = ctx.run(cmd.format(proxy_cmd=""), hide=(not verbose), warn=allow_fail)
-        if not res.ok:
+        if res is None or not res.ok:
             error(f"[-] Failed: {cmd}")
             if allow_fail:
                 return False
-            print_failed(res.stderr)
+            if res is not None:
+                print_failed(res.stderr)
             raise Exit("command failed")
 
         return True
 
     @staticmethod
-    def move_to_shared_directory(ctx, _, source, subdir=None):
+    def move_to_shared_directory(ctx: Context, _: 'HostInstance', source: str, subdir: Optional[PathOrStr] = None):
```

Comment:
What is the difference between `Optional[Union[PathLike, str]` and `Optional[PathOrStr]`?

Reply:
Nothing, `PathOrStr` is defined as an alias of `Union[PathLike, str]`, it's just for convenience.

Reply:
Why don't we use `PathOrStr` everywhere?

Reply:
I didn't notice there was one instance missing the change, done now :D